### PR TITLE
Added catalyst and tvOS support for SPM integration tests

### PR DESCRIPTION
**Pull request by aboedo**
#1162 highlights the fact that we're not testing for macOS compatibility in SPM integration tests, so stuff could fall through the cracks. 

This adds `catalyst` and `tvOS` support so that tests would fail if we've messed up macOS compatibility on SPM.
See https://app.circleci.com/pipelines/github/RevenueCat/purchases-ios/4883/workflows/358209fb-3643-406e-987d-24325ed18ac1/jobs/15354

# TODO:
- [x] Disable integration tests when not on a release branch.

### CODE REVIEW

File: IntegrationTests/SPMIntegration/fastlane/Scanfile [hunk 1]
```diff
@@ -0,0 +1 @@
+catalyst_platform('macos')
```
Comment (aboedo) [inline IntegrationTests/SPMIntegration/fastlane/Scanfile:1]: same here, temporary for testing but we should probably not keep it, we should test both iOS and macOS

File: .circleci/config.yml [hunk 2]
```diff
@@ -359,7 +359,7 @@ workflows:
               # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
               ignore: /pull\/[0-9]+/
       - integration-tests-cocoapods: *release-tags-and-branches
-      - integration-tests-swift-package-manager: *release-tags-and-branches
+      - integration-tests-swift-package-manager
```
Comment (NachoSoto) [inline .circleci/config.yml:362]: Note to self: undo this change before merging

**Comment by NachoSoto** [comment]
Working on this today and/or tomorrow

**Comment by NachoSoto** [comment]
This should work now?

**Comment by aboedo** [comment]
I can't approve because I opened the PR originally, but feel free to approve and merge!

**Comment by NachoSoto** [comment]
bekphnqftcb41

**Review by NachoSoto — APPROVED** [review]
